src/main/java/org/jabref/logic/preferences/TimestampPreferences.java

```diff
@@ -2,26 +2,40 @@
 
 import java.time.LocalDateTime;
 import java.time.format.DateTimeFormatter;
+import java.time.temporal.ChronoUnit;
+import java.util.Objects;
 
 import org.jabref.model.entry.field.Field;
+import org.jabref.model.entry.field.StandardField;
 
 public class TimestampPreferences {
-    private final boolean useTimestamps;
+    private final boolean addCreationDate;
+    private final boolean addModificationDate;
+
+    // Old settings used for migration
     private final boolean updateTimestamp;
     private final Field timestampField;
     private final String timestampFormat;
-    private final boolean overwriteTimestamp;
 
-    public TimestampPreferences(boolean useTimestamps, boolean updateTimestamp, Field timestampField, String timestampFormat, boolean overwriteTimestamp) {
-        this.useTimestamps = useTimestamps;
+    public TimestampPreferences(boolean addCreationDate, boolean modifyTimestamp, boolean updateTimestamp, Field timestampField, String timestampFormat) {
+        this.addCreationDate = addCreationDate;
+        this.addModificationDate = modifyTimestamp;
         this.updateTimestamp = updateTimestamp;
-        this.timestampField = timestampField;
+        this.timestampField = Objects.isNull(timestampField) ? StandardField.TIMESTAMP : timestampField;
```

Comment:
Can this distinction be done on the caller side? The WhateverPreferences-objects schouldn't be much more than data holders.

Reply:
Done!

Reply:
Forgot to push?

Reply:
Yeah, it's in now ^^